docs/devel.md

```diff
@@ -47,3 +59,38 @@ spec:
     - pathPrefix: /etc/docker
 ```
 
+# Testing
+
+What we cover here:
+
+- workflow
+- design/goals
+- why do we use travis and circleci 
+
+## Workflow:
+
+* e2e-tests: 
+`make e2e-tests` will run e2e tests. This tests are run also in CI with `kind`.
+
+We assume that you have already have a running cluster with the `KUBECONFIG` env variable set.
+You can use `kind` for deploying a cluster or another method of your choice.
+
+## Key design and goals:
+
+### Local == Remote
+
+This project use everywhere Make targets.
+In this way when we run `make test` locally or on CI remotely, there is no difference.
+The goal is that for an user/dev should be not differnce to run locally or what is run by CI.
+
+### Makefile is the api
+
+As stated before, the Makefile targets is where other integration tools are dependent for.
+
+In this way we can remove/change the internal logic without modifying the API, and braeking external deps.
+
+## Why do we use Travis and circleCI?
+
+The golden rule is: We use Travis only for `end to end tests` and for the rest we use circleci.
+Since `kind` doesn't workf in CircleCI because of network isolation, we use travis only for `e2e` tests.
+If you want to add other tests that aren't e2e, use Circleci. ( but remember that we use makefile targets, so we don't really add much to this, in this way for use travis/circleci are just plugin/plugout logic).
```

Comment:
I don't get this sentence. The tests should be independent of the environment. What you mean by 'adding other tests'? This is actually contradictory to the next sentence: 'we use makefile targets'

Reply:
yop removing. thx